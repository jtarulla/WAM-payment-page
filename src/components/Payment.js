import React, { Component } from 'react';
import page from '../assets/page-2.png';
import card from '../assets/card.png';
import cardsNet from '../assets/card_networks.png';
import payButton from '../assets/buttons/btn-pay.png';
import backButton from '../assets/buttons/btn-back.png';

export class Payment extends Component {
	continue = e => {
		e.preventDefault();
		this.props.nextStep();
	};

	back = e => {
		e.preventDefault();
		this.props.prevStep();
	};

	render() {
		return (
			<>
				<div className='page-number'>
					<img src={page} alt='page-number' />
				</div>
				<div className='container'>
					<h2>Pago del producto</h2>
					<p>
						Una vez efectuado el pago del producto, recibirás un email con los
						detalles de la compra.
					</p>
					<div className='card-area'>
						<div className='card'>
							<img src={card} alt='card-networks' />
						</div>
						<h3>Tarjeta de crédito</h3>
						<p>
							Se redirigirá a una plataforma de pago. Se trata de un proceso
							seguro. Puede que la validación del pago tarde 24 horas en
							completarse.
						</p>
						<div className='card-net'>
							<img src={cardsNet} alt='card-networks' />
						</div>
					</div>
					<div className='btn-back' onClick={this.back}>
						<img src={backButton} alt='back-button' />
					</div>
					<div className='btn-continue' onClick={this.continue}>
						<img src={payButton} alt='pay-button' />
					</div>
				</div>
			</>
		);
	}
}

export default Payment;
