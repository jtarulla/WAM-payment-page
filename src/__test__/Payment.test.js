import { shallow, mount } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import Payment from '../components/Payment';

describe('Payment component', () => {
	const props = {
		nextStep: jest.fn(),
		prevStep: jest.fn()
	};

	it('renders without error given the required props', () => {
		const wrapper = shallow(<Payment {...props} />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('should call nextStep prop when clicking [.btn-continue]', () => {
		const wrapper = mount(<Payment {...props} />);
		const button = wrapper.find('.btn-continue');
		button.simulate('click', {
			preventDefault: () => {}
		});
		const callback = props.nextStep.mock.calls.length;
		expect(callback).toBe(1);
		expect(props.nextStep).toHaveBeenCalled();
	});

	it('should call prevStep prop when clicking [.btn-back]', () => {
		const wrapper = mount(<Payment {...props} />);
		const button = wrapper.find('.btn-back');
		button.simulate('click', {
			preventDefault: () => {}
		});
		const callback = props.prevStep.mock.calls.length;
		expect(callback).toBe(1);
		expect(props.prevStep).toHaveBeenCalled();
	});
});
