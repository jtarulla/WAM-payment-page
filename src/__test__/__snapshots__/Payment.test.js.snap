// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment component renders without error given the required props 1`] = `
<Fragment>
  <div
    className="page-number"
  >
    <img
      alt="page-number"
      src="page-2.png"
    />
  </div>
  <div
    className="container"
  >
    <h2>
      Pago del producto
    </h2>
    <p>
      Una vez efectuado el pago del producto, recibirás un email con los detalles de la compra.
    </p>
    <div
      className="card-area"
    >
      <div
        className="card"
      >
        <img
          alt="card-networks"
          src="card.png"
        />
      </div>
      <h3>
        Tarjeta de crédito
      </h3>
      <p>
        Se redirigirá a una plataforma de pago. Se trata de un proceso seguro. Puede que la validación del pago tarde 24 horas en completarse.
      </p>
      <div
        className="card-net"
      >
        <img
          alt="card-networks"
          src="card_networks.png"
        />
      </div>
    </div>
    <div
      className="btn-back"
      onClick={[Function]}
    >
      <img
        alt="back-button"
        src="btn-back.png"
      />
    </div>
    <div
      className="btn-continue"
      onClick={[Function]}
    >
      <img
        alt="pay-button"
        src="btn-pay.png"
      />
    </div>
  </div>
</Fragment>
`;
